### PR TITLE
exit with usage when need login to image registry

### PR DESCRIPTION
**Pull request by medyagh**
closes https://github.com/kubernetes/minikube/issues/8124

### before this PR
 if user needed logging into the regsitry, (in this case github packages) there would be 
long error lines and crash 

```
😄  minikube v1.10.1 on Ubuntu 20.04
✨  Using the docker driver based on existing profile
👍  Starting control plane node minikube in cluster minikube
🚜  Pulling base image ...
E0520 12:59:32.299496  159997 cache.go:129] Error downloading kic artifacts:  getting remote image: GET https://docker.pkg.github.com/v2/kubernetes/minikube/kicbase/manifests/v0.0.10: UNAUTHORIZED: GitHub Docker Registry needs login
🤷  docker "minikube" container is missing, will recreate.
E0520 12:59:32.526096  159997 oci.go:79] docker daemon seems to be stuck. Please try restarting your docker. Will try to delete anyways: unknown state "minikube": docker inspect minikube --format={{.State.Status}}: exit status 1
stdout:


stderr:
Template parsing error: template: :1:8: executing "" at <.State.Status>: map has no entry for key "State"
🔥  Creating docker container (CPUs=2, Memory=3900MB) ...
🤦  StartHost failed, but will try again: recreate: creating host: create: creating: create kic node: create container: docker run -d -t --privileged --security-opt seccomp=unconfined --security-opt apparmor=unconfined --tmpfs /tmp --tmpfs /run -v /lib/modules:/lib/modules:ro --hostname minikube --name minikube --label created_by.minikube.sigs.k8s.io=true --label name.minikube.sigs.k8s.io=minikube --label role.minikube.sigs.k8s.io= --label mode.minikube.sigs.k8s.io=minikube --volume minikube:/var --cpus=2 --memory=3900mb -e container=docker --expose 8443 --publish=127.0.0.1::8443 --publish=127.0.0.1::22 --publish=127.0.0.1::2376 --publish=127.0.0.1::5000 docker.pkg.github.com/kubernetes/minikube/kicbase:v0.0.10: exit status 125
stdout:

stderr:
Unable to find image 'docker.pkg.github.com/kubernetes/minikube/kicbase:v0.0.10' locally
docker: Error response from daemon: Get https://docker.pkg.github.com/v2/kubernetes/minikube/kicbase/manifests/v0.0.10: no basic auth credentials.
See 'docker run --help'.

🤷  docker "minikube" container is missing, will recreate.
E0520 13:00:02.871907  159997 oci.go:79] docker daemon seems to be stuck. Please try restarting your docker. Will try to delete anyways: unknown state "minikube": docker inspect minikube --format={{.State.Status}}: exit status 1
stdout:


stderr:
Template parsing error: template: :1:8: executing "" at <.State.Status>: map has no entry for key "State"
...
...
...

```

After this PR: exit short and sweet and provide link to how to login to the fall back image repos

## After this PR:

```
😄  minikube v1.10.1 on Ubuntu 20.04
✨  Using the docker driver based on existing profile
👍  Starting control plane node minikube in cluster minikube
🚜  Pulling base image ...
📶  Unfortunately, could not download the base image gcr.io/k8s-minikube/kicbase1:v0.0.10 
❗  In order to use the fall back image, you need to log in to the github packages registry
📘  Please visit the following link for documentation around this: 
        https://help.github.com/en/packages/using-github-packages-with-your-projects-ecosystem/configuring-docker-for-use-with-github-packages#authenticating-to-github-packages

💡  Please either authenticate the registry or use --base-image option to use a different registry.

```

### CODE REVIEW

File: pkg/minikube/node/cache.go [hunk 1]
```diff
@@ -126,8 +127,21 @@ func beginDownloadKicArtifacts(g *errgroup.Group, cc *config.ClusterConfig) {
 // WaitDownloadKicArtifacts blocks until the required artifacts for KIC are downloaded.
 func waitDownloadKicArtifacts(g *errgroup.Group) {
 	if err := g.Wait(); err != nil {
-		glog.Errorln("Error downloading kic artifacts: ", err)
-		return
+		if err != nil {
+			if errors.Is(err, image.ErrGithubNeedsLogin) {
+				glog.Warningf("Error downloading kic artifacts: %v", err)
+				out.T(out.Connectivity, "Unfortunately, could not download the base image {{.image_name}} ", out.V{"image_name": strings.Split(kic.BaseImage, "@")[0]})
+				out.WarningT("In order to use the fall back image, you need to log into to the github packages registry")
```
Comment (priyawadhwa) [inline pkg/minikube/node/cache.go:134]: you have the word 'to' twice -- replace "log into to" with "log in to"

File: pkg/minikube/node/cache.go [hunk 2]
```diff
@@ -126,8 +127,21 @@ func beginDownloadKicArtifacts(g *errgroup.Group, cc *config.ClusterConfig) {
...
+				out.T(out.Documentation, "https://help.github.com/en/packages/using-github-packages-with-your-projects-ecosystem/configuring-docker-for-use-with-github-packages#authenticating-to-github-packages")
+			}
+			if errors.Is(err, image.ErrGithubNeedsLogin) || errors.Is(err, image.ErrNeedsLogin) {
+				exit.UsageT(`Please either authenticate the registry or use --base-image option to use a different registry.`)
```
Comment (priyawadhwa) [inline pkg/minikube/node/cache.go:138]: Please either authenticate **to** the registry or use the --base-image **flag** to use an image from a different registry

File: pkg/minikube/node/cache.go [hunk 3]
```diff
@@ -126,8 +127,21 @@ func beginDownloadKicArtifacts(g *errgroup.Group, cc *config.ClusterConfig) {
 // WaitDownloadKicArtifacts blocks until the required artifacts for KIC are downloaded.
 func waitDownloadKicArtifacts(g *errgroup.Group) {
 	if err := g.Wait(); err != nil {
-		glog.Errorln("Error downloading kic artifacts: ", err)
-		return
+		if err != nil {
+			if errors.Is(err, image.ErrGithubNeedsLogin) {
+				glog.Warningf("Error downloading kic artifacts: %v", err)
+				out.T(out.Connectivity, "Unfortunately, could not download the base image {{.image_name}} ", out.V{"image_name": strings.Split(kic.BaseImage, "@")[0]})
```
[the remark this answers is not in the document]
Comment (medyagh) [reply pkg/minikube/node/cache.go:133]: done

File: pkg/minikube/node/cache.go (hunk 1 above)
Comment (medyagh) [reply pkg/minikube/node/cache.go:134]: done

File: pkg/minikube/node/cache.go [hunk 4]
```diff
@@ -126,8 +127,21 @@ func beginDownloadKicArtifacts(g *errgroup.Group, cc *config.ClusterConfig) {
...
+				out.WarningT("In order to use the fall back image, you need to log into to the github packages registry")
+				out.T(out.Documentation, "https://help.github.com/en/packages/using-github-packages-with-your-projects-ecosystem/configuring-docker-for-use-with-github-packages#authenticating-to-github-packages")
```
[the remark this answers is not in the document]
Comment (medyagh) [reply pkg/minikube/node/cache.go:135]: done

**Comment by afbjorklund** [comment]
It will still fail, though. Just with a different error message.

It would be better to use a more public image mirror....

**Comment by ligh0721** [comment]
still fail.

```shell
$ docker login https://docker.pkg.github.com                                                                                                                 1 
Username (xxxxx): 
Password: 
Login Succeeded

$ minikube start --driver=docker
* minikube v1.11.0 on Centos 7.6.1810
* Using the docker driver based on existing profile
* Starting control plane node minikube in cluster minikube
* Pulling base image ...
* Unfortunately, could not download the base image gcr.io/k8s-minikube/kicbase:v0.0.10 
! In order to use the fall back image, you need to log in to the github packages registry
* Please visit the following link for documentation around this: 
        https://help.github.com/en/packages/using-github-packages-with-your-projects-ecosystem/configuring-docker-for-use-with-github-packages#authenticating-to-github-packages

* Please either authenticate to the registry or use --base-image flag to use a different registry.
```

**Pull request merged**